### ISSUE TITLE
Improper Assembly Crash in Naked Hook

### ISSUE THREAD

**Issue by maatko** [SystematicSkid/java-jit-hooks#8]
[hs_err_pid10364.log](https://github.com/user-attachments/files/22779968/hs_err_pid10364.log)

**Comment by SystematicSkid** [comment]
If you are using the same version as me, you can try out the original commit from the PoC to test.

**Comment by maatko** [comment]
In the following commit fc09a7cb2b4ef3b85916854aef801356a1674933 the issue starts occuring.

This is the latest working commit i found 509f98470cac81992baf68f660020adaf4da97d6

**Comment by SystematicSkid** [comment]
Yeah I think I know why. I have this fixed on the second project I am working on, and I will port it over soon.
